### PR TITLE
Add table to rich text fields

### PR DESCRIPTION
**Pull request by ValerieCodeurs**

### CODE REVIEW

File: src/core/shape/RichTextShape.ts [hunk 1]
```diff
@@ -36,7 +36,11 @@ export enum RichTextElements {
   br = 'br',
   small = 'small',
   sup = 'sup',
-  sub = 'sub'
+  sub = 'sub',
+  table = 'table',
+  tableCell = 'td',
```
Comment (benmerckx) [inline src/core/shape/RichTextShape.ts:41]: You can use the html tag names here

File: src/ui/RichText.tsx [hunk 2]
```diff
@@ -129,7 +137,12 @@ export type RichTextProps<Blocks extends Schema> = {
   text?: ComponentType<{children: string}>
 } & {
   [K in keyof typeof RichTextElements]?:
-    | ComponentType<JSX.IntrinsicElements[K]>
+    | ComponentType<
+        React.JSX.IntrinsicElements[Extract<
```
Comment (benmerckx) [inline src/ui/RichText.tsx:141]: This change is not needed if we use html tag names in RichTextElements

File: src/field/richtext/RichTextToolbar.tsx [hunk 3]
```diff
@@ -164,6 +166,67 @@ export const RichTextToolbar = forwardRef(function RichTextToolbar(
               </DropdownMenu.Item>
             </DropdownMenu.Items>
           </DropdownMenu.Root>
+          <DropdownMenu.Root top>
```
Comment (benmerckx) [inline src/field/richtext/RichTextToolbar.tsx:169]: Can we make this optional by adding an `enableTables?: boolean` to [`RichTextOptions`](https://github.com/alineacms/alinea/blob/d65a1dcdd8a2b0635ed11bd22a1480f436e22a09/src/field/richtext/RichTextField.tsx#L8)?

Comment (ValerieCodeurs) [reply src/field/richtext/RichTextToolbar.tsx]: [(https://github.com/alineacms/alinea/pull/381/commits/cb1e41f5b99087898f8c51145b3ec4ff4db1fb09)]

File: src/ui/RichText.tsx [hunk 4]
```diff
@@ -129,7 +137,7 @@ export type RichTextProps<Blocks extends Schema> = {
   text?: ComponentType<{children: string}>
 } & {
   [K in keyof typeof RichTextElements]?:
-    | ComponentType<JSX.IntrinsicElements[K]>
+    | ComponentType<JSX.IntrinsicElements>
```
Comment (benmerckx) [inline src/ui/RichText.tsx:140]: We still need to use the original type here: | ComponentType<JSX.IntrinsicElements[K]>

**Review by benmerckx — CHANGES_REQUESTED** [review]

File: src/core/shape/RichTextShape.ts (hunk 1 above)
Comment (ValerieCodeurs) [reply src/core/shape/RichTextShape.ts]: Is resolved.

File: src/ui/RichText.tsx (hunk 2 above)
Comment (ValerieCodeurs) [reply src/ui/RichText.tsx]: Is resolved.

File: src/ui/RichText.tsx (hunk 4 above)
Comment (ValerieCodeurs) [reply src/ui/RichText.tsx]: Change has been reverted.

**Pull request merged**